### PR TITLE
OpenStack: set unique cluster names

### PR DESCRIPTION
**Pull request by Fedosin**
This commit adds the job name hash as a suffix to the cluster name to make it unique.

### CODE REVIEW

**Comment by dprince** [comment]
Yes, this should fix the issues we see with regards to race conditions when multiple openstack jobs run at once.

/lgtm

**Comment by mandre** [comment]
Won't this go over the 14 char limit we've set?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 1]
```diff
@@ -284,6 +284,11 @@ objects:
         cp "$(command -v openshift-install)" /tmp
         mkdir /tmp/artifacts/installer
 
+        # We have to truncate cluster name to 14 chars, because there is a limitation in the install-config
+        # Now it looks like "ci-op-rl6z646h-65230".
+        # We will remove "ci-op-" prefix from there to keep just last 14 characters.
+        export CLUSTER_NAME=${CLUSTER_NAME: -14}
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: The usual IPI template [uses][1] the `${NAMESPACE}-${JOB_NAME_HASH}` you're moving to here.  But we haven't had to truncate there.  Can you link the code that enforces the 14-char limitation?

[1]: https://github.com/openshift/release/blob/65a222271fe372a7e010d89071a6ef8d1b5ef386/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L105

Comment (iamemilio) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: That is specific to the openstack platform. https://github.com/openshift/installer/blob/master/docs/user/openstack/known-issues.md#long-cluster-names

**Pull request merged**